### PR TITLE
Add ipns domain name support

### PR DESCRIPTION
**Pull request by dospore**
As per https://github.com/ensdomains/ens-app/issues/1430 re-add support for domain ipns entries. I saw a couple of comments in the code mentioning deprecation of ipns-ns domain support but wasnt 100% sure so thought Id make the PR anyway to get the conversation started.

I needed this functionality for a personal project I was setting up so ran this locally to set my domain.

This PR goes alongside a PR into ens-ui.

https://www.loom.com/share/d7cf8930936d4b7182b5304703a195cc

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -3315,324 +3311,326 @@ exports.write = function (buffer, value, offset, isLE, mLen, nBytes) {
...
+     * does what `window.btoa` of web browsers do.
+     * @param {String} bin binary string
+     * @returns {string} Base64-encoded string
+     */
+    var _btoa = _hasbtoa ? function (bin) { return btoa(bin); }
+        : _hasBuffer ? function (bin) { return Buffer.from(bin, 'binary').toString('base64'); }
+            : btoaPolyfill;
+    var _fromUint8Array = _hasBuffer
+        ? function (u8a) { return Buffer.from(u8a).toString('base64'); }
+        : function (u8a) {
```
Comment (lidel) [inline dist/index.js:3411]: What is the purpose of changes like this? Should be a separate PR. It hides what was actually changed here, making PR unnecessarily hard to review.

**Review by lidel — CHANGES_REQUESTED** [review]
I don't believe allowing non-cryptographic identifiers should be allowed, but leaving it up to @makoto to make the final call.

My rationale:
- If you use `/ipns/{libp2p-key}` then you have cryptographic chain of trust from ENS chain record, via a signed IPNS record, down to the final `/ipfs/{cid}` content path
- If you use `/ipns/{icann-domain}` in ENS record, then you dont have cryptographic chain of trust: at any point ICANN, or the operator of your TLD  can censor you and change the TXT record without your involvement.
  - Allowing this opens security attack record, and further erodes trust in the premise of ENS being cryptographically-backed  alternative to ICANN.

**Comment by makoto** [comment]
Closing this. Please refer to https://github.com/ensdomains/ui/pull/164#issuecomment-1131514097 for the rationale

File: dist/index.js (hunk 1 above)
Comment (dospore) [reply dist/index.js:3411]: this is just build generated? Would of dropped the dist changes but closed PR due to above.

**Comment by dospore** [comment]
> I don't believe allowing non-cryptographic identifiers should be allowed, but leaving it up to @makoto to make the final call.
> 
> My rationale:
> 
> * If you use `/ipns/{libp2p-key}` (already allowed) then you have cryptographic chain of trust from ENS chain record, via a signed IPNS record, down to the final `/ipfs/{cid}` content path
> * If you allow `/ipns/{icann-domain}` in ENS record (proposed in this PR), then you dont have cryptographic chain of trust: at any point ICANN, or the operator of your TLD  can censor you and change the TXT record without your involvement.
>   
>   * Allowing this opens security attack record, and further erodes trust in the premise of ENS being cryptographically-backed  alternative to ICANN.

Makes sense and is a valid justification. Interested to discuss though, would this not be a /ipns/ dns_link attack vector in general? If we assume TLD operators are willing and able to override your TXT records. Maybe is a discussion to have with their dev team since its unrelated to ENS.
